plugins/modules/ranger_policy.py

```diff
@@ -0,0 +1,160 @@
+#! /usr/bin/env python
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+# -*- coding: utf-8 -*-
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.common._collections_compat import Mapping, Sequence
+from ansible.module_utils.six import iteritems, string_types, text_type
+from ansible.module_utils.six.moves.urllib.parse import quote as urlquote
+from ansible.module_utils.urls import fetch_url, url_argument_spec
+
+def ranger_json_deep_equal(a, b):
+    """Custom comparison of two objects.
+    If objects are Mapping (dict) and if B has a missing key from A return False. If B has additional keys returns True.
+    Ranger Admin can return JSON with additional keys and we should ignore it.
+    """
+    if isinstance(a, Mapping) and isinstance(b, Mapping):
+        for key_a, value_a in iteritems(a):
+            if key_a not in b:
+                return False
+            if not ranger_json_deep_equal(value_a, b[key_a]):
+                return False
+    # "string_types" is a "Sequence", to avoid infinite recursion perform direct comparison
+    elif isinstance(a, string_types) and isinstance(b, string_types):
+        # Convert to unicode before comparison
+        return text_type(a) == text_type(b)
+    elif isinstance(a, Sequence) and isinstance(b, Sequence):
+        if len(a) != len(b):
+            return False
+        for value_a, value_b in zip(a, b):
+            if not ranger_json_deep_equal(value_a, value_b):
+                return False
+    return a == b
+
+def dict_del_key(a, b):
+    """Remove keys in B that are not in A recursively"""
+    if isinstance(a, Mapping) and isinstance(b, Mapping):
+        for key_b in list(b.keys()):
+            if key_b not in a:
+                del b[key_b]
+            else:
+                dict_del_key(a[key_b], b[key_b])
+    # "string_types" is a "Sequence", to avoid infinite recursion return
+    elif isinstance(a, string_types) and isinstance(b, string_types):
+        return
+    elif isinstance(a, Sequence) and isinstance(b, Sequence):
+        for value_a, value_b in zip(a, b):
+            dict_del_key(value_a, value_b)
+
+def main():
+    argument_spec = url_argument_spec()
+    argument_spec.update(
+        policy_mgr_url=dict(type='str', required=True),
+        policy=dict(type='dict', required=True),
+        state=dict(type='str', choices=['present', 'absent']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    policy_mgr_url = module.params['policy_mgr_url']
+    policy = module.params['policy']
+    state = module.params['state'] or 'present'
```

Comment:
```suggestion
    state = module.params.get('state', 'present')
```

Does this work ? At first glance, it would throw a KeyError exception no ?

Reply:
It works with:
```
- name: Configure Ranger Admin policies
  tosit.tdp.ranger_policy:
    policy_mgr_url: "https://{{ groups['ranger_admin'][0] | tosit.tdp.access_fqdn(hostvars) }}:6182"
    policy: "{{ item }}"
    url_username: admin
    url_password: "{{ ranger_admin_password }}"
    force_basic_auth: yes
  loop: "{{ ranger_policies }}"
  loop_control:
    label:
      name: "{{ item.name | default('missing name') }}"
      service: "{{ item.service | default('missing service') }}"
```
Ansible set `None` if `state` is missing.

Reply:
Ok